### PR TITLE
Feature/add ssm param util code

### PR DESCRIPTION
**Pull request by THOR300**
# Description

This pull request adds some code that will be used regularly in prefect flows into the sdk with tests to allow for instantiation of a `VespaSearchAdapter` from aws secrets. This function may be used by all flows that wish to access vespa by reading credentials from ssm. 

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [ ] Skip auto-tagging
- [X] Patch
- [ ] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail (integrated soon)
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [X] New feature
- [ ] Breaking change

## How Has This Been Tested?

Please describe the tests that you added to verify your changes.

## Before submitting

- [X] I've read and followed all steps in the [Making a pull request](https://github.com/climatepolicyradar/cpr-sdk/blob/main/.github/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [X] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/climatepolicyradar/cpr-sdk/blob/main/.github/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [X] If this PR fixes a bug, I've added a test that will fail without my fix.
- [X] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

### CODE REVIEW

File: src/cpr_sdk/ssm.py
Comment (olaughter) [inline src/cpr_sdk/ssm.py]: I'm uneasy about this. On the one hand I can see this being great for programmatic access, we can stop passing in these certs env variables. Putting this in the sdk though makes it seem like its an option for anyone to use. Leading to potentially any dev having write access to prod as standard. which none of us need or should have.

File: src/cpr_sdk/ssm.py
Comment (olaughter) [inline src/cpr_sdk/ssm.py:8]: I wonder if this will ever be used outside of a prefect flow, in which case perhaps we could start a prefect submodule in the sdk that could later have other things, like a deployments wrapper, etc. What do you think?

File: tests/test_ssm.py
Comment (olaughter) [inline tests/test_ssm.py]: I did not know that `tmpdir` was a generally available [fixture](https://docs.pytest.org/en/6.2.x/tmpdir.html#the-tmpdir-fixture)! Very happy to find that out

File: src/cpr_sdk/ssm.py
Comment (THOR300) [reply src/cpr_sdk/ssm.py:8]: No strong opinion, personally I've not seen enough reusable code to warrant that just yet.

File: src/cpr_sdk/ssm.py
Comment (THOR300) [reply src/cpr_sdk/ssm.py]: I hear your concern, someone could assume their prod aws profile and call this function and have access.

We're currently storing secrets in systems manager, this doesn't allow you to provide access permissions on the secret object itself. We'd have to limit access at the iam role level (admin and power user profiles). 

Therefore, I'm going to suggest we take a look at using AWS Secrets instead, this allows us to set permissions on the secret object itself. So we can enforce something like "only this service can access this secret" to block access.

Comment (THOR300) [reply src/cpr_sdk/ssm.py]: We've spoken offline, decided to hold off on merging this function until we have secrets with RBAC and duplicate the code in repos where it's used.

**Review by olaughter — APPROVED** [review]

**Pull request merged**
